addons/hr_expense/models/product_product.py

```diff
@@ -24,3 +24,33 @@ def _compute_standard_price_update_warning(self):
                             "There are unsubmitted expenses linked to this category. Updating the category cost will change expense amounts. "
                             "Make sure it is what you want to do."
                         )
+
+    def write(self, vals):
+        res = super().write(vals)
+        if 'standard_price' not in vals:  # Only check for expenses if the standard price has been updated
+            return res
+
+        expenses_sudo = self.env['hr.expense'].sudo().search([
+            ('state', 'in', ['draft', 'reported']),
+            ('product_id', 'in', self.ids),
+        ])
+        for expense_sudo in expenses_sudo:
+            expense_product = expense_sudo.product_id.sudo(False)
+            tax_domain = self.env['account.tax']._check_company_domain(expense_sudo.company_id)
+            product_has_cost = not expense_sudo.company_currency_id.is_zero(expense_product.standard_price)
+            expense_vals = {
+                'product_has_cost': product_has_cost,
+                'product_has_tax': bool(expense_product.supplier_taxes_id.filtered_domain(tax_domain)),
```

Comment:
Does this filter really have an impact ? :thinking: It will only take the taxes accepted by the current companies anyway, right  ? (because of the sudo(False))